clients/java/src/test/java/io/zeebe/client/impl/worker/JobWorkerImplTest.java

```diff
@@ -0,0 +1,238 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl.worker;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcCleanupRule;
+import io.zeebe.client.ZeebeClient;
+import io.zeebe.client.api.worker.JobHandler;
+import io.zeebe.client.impl.ZeebeClientBuilderImpl;
+import io.zeebe.client.impl.ZeebeClientImpl;
+import io.zeebe.gateway.protocol.GatewayGrpc;
+import io.zeebe.gateway.protocol.GatewayGrpc.GatewayImplBase;
+import io.zeebe.gateway.protocol.GatewayOuterClass;
+import io.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsRequest;
+import io.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsResponse;
+import io.zeebe.gateway.protocol.GatewayOuterClass.ActivatedJob;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.awaitility.Awaitility;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class JobWorkerImplTest {
+
+  private static final JobHandler NOOP_JOB_HANDLER = (client, job) -> {};
+  private static final long SLOW_POLL_DELAY_IN_MS = 1_000L;
+  private static final Duration SLOW_POLL_THRESHOLD = Duration.ofMillis(SLOW_POLL_DELAY_IN_MS / 2);
+
+  @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  private MockedGateway gateway;
+  private ZeebeClient client;
+
+  @Before
+  public void setup() throws IOException {
+    // Generate a unique in-process server name.
+    final String serverName = InProcessServerBuilder.generateName();
+
+    // Create a server, add service, start, and register for automatic graceful shutdown.
+    gateway = new MockedGateway();
+    grpcCleanup.register(
+        InProcessServerBuilder.forName(serverName)
+            .directExecutor()
+            .addService(gateway)
+            .build()
+            .start());
+
+    // Create a client channel and register for automatic graceful shutdown.
+    final ManagedChannel channel =
+        grpcCleanup.register(InProcessChannelBuilder.forName(serverName).directExecutor().build());
+
+    // Create the Zeebe Client
+    client =
```

Comment:
I don't understand what you mean. client is of type `ZeebeClient` which is an interface. What am I missing?

Reply:
Yes but we create it directly, instead of the usual `ZeebeClient.newBuilder().build()`, is what I meant. Though I guess here we have to due to passing the stub, right?

Reply:
Yes, it's the only way to pass the stub in.